src/renderers/shared/fiber/__tests__/ReactIncremental-test.js

```diff
@@ -115,4 +115,54 @@ describe('ReactIncremental', function() {
 
   });
 
+  it('can cancel partially rendered work and restart', function() {
+
+    var ops = [];
+
+    function Bar(props) {
+      ops.push('Bar');
+      return <div>{props.children}</div>;
+    }
+
+    function Foo(props) {
+      ops.push('Foo');
+      return (
+        <div>
+          <Bar>{props.text}</Bar>
+          <Bar>{props.text}</Bar>
+        </div>
+      );
+    }
+
+    // Init
+    ReactNoop.render(<Foo text="foo" />);
+    ReactNoop.flush();
+
+    ops = [];
+
+    ReactNoop.render(<Foo text="bar" />);
+    // Flush part of the work
+    ReactNoop.flushLowPri(20);
```

Comment:
No, this side-effect be queued and discarded. This is what my post in React Core Contributors today was about.